160-select-input/app.R

```diff
@@ -0,0 +1,31 @@
+library(shiny)
+
+shinyApp(
+  ui = fluidPage(
+    p(
+      "This test ensures that", code("selectInput()"), "doesn't introduce extra input variable(s).",
+      a("Issue #2396", href = "https://github.com/rstudio/shiny/issues/2396"), "/",
+      a("PR #2418", href = "https://github.com/rstudio/shiny/pull/2418")
+    ),
+    selectInput("variable", "Variable:",
+                c("Cylinders" = "cyl", "Transmission" = "am", "Gears" = "gear")
+    ),
+    uiOutput("testResult")
+  ),
+  server = function(input, output, session) {
+
+    output$testResult <- renderUI({
```

Comment:
I think this should have a line with just `input$variable`, to ensure that it's updated _after_ the input value is received. Otherwise, it's not necessarily guaranteed to run this code after receiving the input value. (In practice with the way things currently work in Shiny, things do happen in that order, but it's good to have it guaranteed by the code.)